vllm/executor/uniproc_executor.py

```diff
@@ -93,9 +93,10 @@ def _init_executor(self) -> None:
             ("ExecutorWithExternalLauncher needs deterministic "
             "execution, so it"
             "does not support delay_factor in scheduling")
-        assert not envs.VLLM_USE_V1, \
-            ("V1 architecture cannot guarantee deterministic execution, "
-            "so it is not supported in ExecutorWithExternalLauncher.")
+        if envs.VLLM_USE_V1:
+            assert not envs.VLLM_ENABLE_V1_MULTIPROCESSING, \
+            ("To get deterministic execution in V1, "
+            "please set VLLM_ENABLE_V1_MULTIPROCESSING=0")
```

Comment:
A dumb question: Can we somehow set this env automatically when using the launcher?

Reply:
good point, fixed in [1eaf5c7](https://github.com/vllm-project/vllm/pull/13642/commits/1eaf5c7c02085909d16004a200e2d6f695bab8cd)